ts/components/animations/AnimatedRing.tsx

```diff
@@ -0,0 +1,81 @@
+import { Millisecond } from "italia-ts-commons/lib/units";
+import * as React from "react";
+import { Animated, Easing, StyleSheet } from "react-native";
+import themeVariables from "../../theme/variables";
+
+const styles = StyleSheet.create({
+  ring: {
+    position: "absolute",
+    borderWidth: 2,
+    borderColor: themeVariables.brandPrimary
+  }
+});
+
+interface Props {
+  dimension: number;
+  duration: Millisecond;
+  interval: Millisecond;
+  boxDimension: number;
+}
+/**
+ * Create a ring with opacity and scale effect with the primary color
+ */
+export default class AnimatedRing extends React.Component<Props> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {};
+  }
+
+  private animatedValue = new Animated.Value(0);
+
+  public componentDidMount() {
+    setTimeout(() => {
```

Comment:
It could be happen that the users leave the screen before the scheduled function in setTimeout is triggered.
To avoid this you should take care of the handler of setTimeout and then use it to clear when you leave the screen
Here some pseudo-code

```typescript
public componentDidMount() {
   const idSetTimeout = setTimeout(...)
   this.setState({idSetTimeout})
...
}

  public componentWillUnmount() {
  clearTimeout(this.state.idSetTimeout);
...
}
```